src/modules/ens/sagas.ts

```diff
@@ -0,0 +1,129 @@
+import { Eth } from 'web3x-es/eth'
+import { call, put, takeEvery, select } from 'redux-saga/effects'
+import {
+  GET_ENS_REQUEST, 
+  GetENSRequestAction, 
+  getENSSuccess, 
+  getENSFailure, 
+  SET_ENS_REQUEST,
+  SetENSRequestAction, 
+  setENSSuccess, 
+  setENSFailure
+} from './actions'
+import { ENS } from 'contracts/ENS'
+import { ENSResolver } from 'contracts/ENSResolver'
+import { Address } from 'web3x-es/address'
+import { getAddress } from 'decentraland-dapps/dist/modules/wallet/selectors'
+import { 
+  ENS_ADDRESS, 
+  ENS_RESOLVER_ADDRESS 
+} from 'modules/common/contracts'
+import { ipfs } from 'lib/api/ipfs'
+import { namehash } from "@ethersproject/hash";
+import * as contentHash from 'content-hash'
+import { ENS_EMPTY_CONTENT, ENS_EMPTY_RESOLVER } from './constants'
+
+export function* ensSaga() {
+  yield takeEvery(GET_ENS_REQUEST, handleGetENSRequest)
+  yield takeEvery(SET_ENS_REQUEST, handleSetENSRequest)
+}
+
+function* handleSetENSRequest(action: SetENSRequestAction) {
+  const { land, ens } = action.payload
+  try {
+    const [eth, from] = yield getEth()
+    const nodehash = namehash(ens)
+    const ensContract = new ENS(eth, Address.fromString(ENS_ADDRESS))
+    const resolverContract = new ENSResolver(eth, Address.fromString(ENS_RESOLVER_ADDRESS))
+
+    const ipfsHash = yield call(() => ipfs.uploadRedirectionFile(land))
+    const hash = contentHash.fromIpfs(ipfsHash)
```

Comment:
What do you think of moving `contentHash.fromIpfs(ipfsHash)` to a utils/helper file? You will be able to return just `0x${hash}` from there which is the only thing that you will end up using. Also, it seems that this piece of code is repeated below.

```javascript
const ipfsHash = yield call(() => ipfs.uploadRedirectionFile(land))
const hash = contentHash.fromIpfs(ipfsHash)
```